storybook/src/admin/form/DependentAsyncSelects.tsx

```diff
@@ -0,0 +1,83 @@
+import { gql, useApolloClient } from "@apollo/client";
+import { AsyncSelectField, FinalForm } from "@comet/admin";
+import { Box } from "@mui/material";
+import { storiesOf } from "@storybook/react";
+import * as React from "react";
+import { OnChange } from "react-final-form-listeners";
```

Comment:
I'm not sure if we should use that dependency

- it's official (in final-form github org)
- but see this issue https://github.com/final-form/react-final-form-listeners/issues/41
- and it's implemented using class components :scream: (ok ok, comet also has a few)

Reply:
You're probably right. The question that remains if we want a custom `OnChange` component or just roll with `<Field subscription={{ value: true }} />`. I tried to port the official OnChange to function components and compare it to a Field change in d47375650f4f8c98b2742fff69c7fe7b8537f662.

One downside of using field is that we don't have information on the previous value, which may result in some false positive calls (see beginning of attached screencast). What do you think?


https://github.com/vivid-planet/comet/assets/48853629/db343cde-e686-4120-800e-f8e803c2e376

Reply:
> roll with <Field subscription={{ value: true }} />

I think that is for performance optimization, but it doesn't guarantee to be called only on change

> I tried to port the official OnChange to function components and compare it to a Field change in https://github.com/vivid-planet/comet/commit/d47375650f4f8c98b2742fff69c7fe7b8537f662.

you might even simplify that by using useField

Reply:
Copied/adapted from react-final-form-listeners in 72c01d45758f05950ea0cd75fcdc9c062e043a50. Should this be a separate PR, or is it okay in this one (since it's needed here)?

Also, should we change the naming and/or API? For instance, `<RegisterFieldChangeListener name="product" onChange=((value, previousValue) => {}) />`

Reply:
> Copied/adapted from react-final-form-listeners in https://github.com/vivid-planet/comet/commit/72c01d45758f05950ea0cd75fcdc9c062e043a50. Should this be a separate PR, or is it okay in this one

No benefit in creating a new PR for such a simple thing.

> Also, should we change the naming and/or API? For instance, <RegisterFieldChangeListener name="product" onChange=((value, previousValue) => {}) />

yes, OnChange in the context of `@comet/admin` is not enough information.

- FinalFormOnChange
- FieldOnChnge
- FinalFormFieldOnChange

Reply:
Renamed to `OnChangeField`. Liked it better than `FieldOnChange` because it "reads" more like sentence.